core/src/main/scala/kafka/network/SocketServer.scala

```
@@ -1220,6 +1227,8 @@ private[kafka] class Processor(
         inflightResponses.remove(connectionId).foreach(updateRequestMetrics)
         // the channel has been closed by the selector but the quotas still need to be updated
         connectionQuotas.dec(listenerName, InetAddress.getByName(remoteHost))
+        // Call listeners to notify for closed connection.
+        connectionDisconnectListeners.foreach(listener => listener.onDisconnect(connectionId))
```

Should we also handle the other path that doesn't update disconnected? connectionQuotas already manages both paths.

https://github.com/apache/kafka/blob/76a1af984b39d9890fe26954aff36bb1a321af77/core/src/main/scala/kafka/network/SocketServer.scala#L1244

---

Additionally, should we swallow the exception in the loop to ensure all listeners are executed?

---

Thanks for pointing out, agree. Done.